Add Mac os x installation scripts

nsryan2 opened the pull request:
This PR adds a macosx.sh file (which is a replica of the the ubuntu_mint.sh) and a osx_10.15.06.sh file (which is a replica of the ubuntu_16.04.sh file). They have been adapted to hopefully work on mac.

bam241 commented on osx_10.15.6.sh:
```diff
@@ -0,0 +1,24 @@
+# Use package manager for as many packages as possible
+brew_package_list="glib python3 wget \
+             git cmake vim emacs gcc openblas \
+             lapack autoconf libtool make"
```
this is missing `eigen`

nsryan2 replied:
Thanks for the catch, is there anything else you noticed missing on the list? I tried to break up the ubuntu packages (like build-essential, python-software-properties, and software-properties-common), but some may have slipped through the cracks.

bam241 replied:
hdf5 is also missing :)

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,147 @@
+    cd moab-repo
+    mkdir -p build
+    cd build
+    cmake ../ -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_libdir} \
```
```suggestion
    cmake ../ -DENABLE_HDF5=ON \
```

this is not required, as hdf5_libdir is not defined, and brew installs it in a system known location

gonuke replied:
`hdf5_libdir` is defined in the script that calls this, so you probably want to remove that line, too

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,143 @@
+eval brew install $brew_package_list
+export PATH="$HOME/.local/bin:$PATH"
+eval pip3 install --user --upgrade pip3
+eval pip3 install --user $pip_package_list
```
```suggestion
eval pip3 install $pip_package_list
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,143 @@
+eval brew update
+eval brew install $brew_package_list
+export PATH="$HOME/.local/bin:$PATH"
+eval pip3 install --user --upgrade pip3
```
```suggestion
eval pip3 install --upgrade pip3
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,142 @@
+eval brew update
+eval brew install $brew_package_list
+export PATH="$HOME/.local/bin:$PATH"
+eval pip3 install $pip_package_list
```
```suggestion
eval sudo pip3 install $pip_package_list
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,142 @@
+    echo "export C_INCLUDE_PATH=${install_dir}/moab/include:\$C_INCLUDE_PATH" >> ~/.bashrc
+
+    PYTHON_VERSION=$(python -c 'import sys; print(sys.version.split('')[0][0:3])')
+    echo "if [ -z \$PYTHONPATH ]" >> ~/.bashrc
```
```suggestion
    echo "if [ -z \"\$PYTHONPATH\" ]" >> ~/.bashrc
```

bam241 replied:
the `-z` test needs quotation mark to work (at least on Mac)

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,142 @@
+    echo "export PATH=$HOME/.local/bin:\$PATH" >> ~/.bashrc
+    echo "export LD_LIBRARY_PATH=$HOME/.local/lib:\$LD_LIBRARY_PATH" >> ~/.bashrc
+    echo "alias build_pyne='python setup.py install --user -- -DMOAB_LIBRARY=$install_dir/moab/lib -DMOAB_INCLUDE_DIR=$install_dir/moab/include'" >> ~/.bashrc
+
```
```suggestion
PYTHON_VERSION=$(python -c 'import sys; print(sys.version.split('')[0][0:3])')
echo "  export PYTHONPATH=~/.local/lib/python${PYTHON_VERSION}/site-packages:\$PYTHONPATH" >> ~/.bashrc
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,142 @@
+
+
+build_moab
+
```
```suggestion

source ~/.bashrc
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,144 @@
+    make install
+    export LD_LIBRARY_PATH=${install_dir}/moab/lib:$LD_LIBRARY_PATH
+    export LIBRARY_PATH=${install_dir}/moab/lib:$LIBRARY_PATH
+    if [ -z \$PYTHONPATH ]
+    then
+        export PYTHONPATH=${install_dir}/moab/lib/python2.7/site-packages
+    else
+        export PYTHONPATH=${install_dir}/moab/lib/python2.7/site-packages:\$PYTHONPATH
+    fi
```
this lines should be removed

bam241 commented:
you also need to change the permission of macosx.sh and ox.????.sh scripts to executable

gonuke commented:
You'll want to exclude the `.DS_Store` file/directory from your PR.  Also might be useful to keep the shell script and dockerfile in different PRs.  I think docker may be even more challenging...

nsryan2 commented:
My bad, I pushed changes to the wrong branch when I changed the permissions so those got wrapped up in it. I thought I deleted that dockerfile.  Thanks for the note

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,135 @@
+mkdir -p $install_dir
+
+build_moab
+
```
```suggestion

build_dagmc
```

bam241 replied:
I think that's the last thing missing.

@gonuke once this is added, I thing this should be ready for final review

gonuke commented on macosx.sh:
```diff
@@ -0,0 +1,136 @@
+              -DCMAKE_INSTALL_PREFIX=${install_dir}/moab
+    make
+    make install
+    
```
Difference from Ubuntu: why not update the ENV Variables directly here (instead of sourcing .bashrc below)?

bam241 replied:
I think we should update the linux version accordingly:

I see 2 advantages:
- simplify the scripts
- allow to check if .bashrc if properly formed

gonuke commented on macosx.sh:
```diff
@@ -0,0 +1,136 @@
+
+    repo_name=$1
+
+    if [ -d $repo_name ] ; then
```
Why no curly braces as there are in the ubuntu script?  They may not be strictly necessary but good practice. (This comment applies in many places throughout.)

bam241 replied:
```suggestion
    if [ -d ${repo_name} ] ; then
```

gonuke commented on macosx.sh:
```diff
@@ -0,0 +1,136 @@
+        TAG=$(git describe --abbrev=0 --tags)
+        git checkout tags/`echo $TAG` -b `echo $TAG`
+    fi
+    python setup.py install --user -- -DMOAB_LIBRARY=$install_dir/moab/lib -DMOAB_INCLUDE_DIR=$install_dir/moab/include
```
This looks like an "old-fashioned" way to configure setup.py.  Why not the `--moab` and `--dagmc` options?  Also, no DAGMC?  Is that not possible?

bam241 replied:
fixing this right now

gonuke commented on macosx.sh:
```diff
@@ -0,0 +1,136 @@
+    cd $install_dir/pyne
+    cd tests
+
+    nosetests .
```
Is this part of the move away from using the `travis-tests`?

bam241 replied:
```suggestion
    travis_travis-run-tests.sh python3
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,144 @@
+    
+
+	python setup.py install --user -- -DMOAB_LIBRARY=${install_dir}/moab/lib
+                                      -DMOAB_INCLUDE_DIR=${install_dir}/moab/include
```
```suggestion
                                      -DMOAB_INCLUDE_DIR=${install_dir}/moab/include \
```

bam241 commented on macosx.sh:
```diff
@@ -0,0 +1,144 @@
+    fi
+    
+
+	python setup.py install --user -- -DMOAB_LIBRARY=${install_dir}/moab/lib
```
```suggestion
	python setup.py install --user -- -DMOAB_LIBRARY=${install_dir}/moab/lib \
```

gonuke commented:
An item to confirm/test for @bam241 or @nsryan2:
1. we currently prevent building of the spatial solvers on macOS, apparently in 2 separate places: https://github.com/pyne/pyne/blob/develop/src/CMakeLists.txt#L67 and https://github.com/pyne/pyne/blob/develop/cmake/PyneMacros.cmake#L151

Do we need both of these, or can we rely on the one in `PyneMacros`?

2. Can we actually support building the spatial solvers on MacOS?  This may be a more challenging question.

nsryan2 commented:
I keep getting the error asking for a stdio.h file, but I'm not sure how to make or install one. This failure happens in the moab part of the install, is there a package or file I can install separately to prevent this?

> /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/stdio.h:107:15: fatal error: 
      'stdio.h' file not found
#include_next <stdio.h>
              ^~~~~~~~~
1 error generated.
error: command '/Library/Developer/CommandLineTools/usr/bin/cc' failed with exit status 1
make[2]: *** [pymoab/CMakeFiles/pymoab-local-install] Error 1
make[1]: *** [pymoab/CMakeFiles/pymoab-local-install.dir/all] Error 2
make: *** [all] Error 2

katyhuff commented:
Double check your installation of xcode's command-line tools. https://github.com/frida/frida/issues/338
